### PR TITLE
[dev 4.0.1 -> staging] Add namespace support for Android

### PR DESCRIPTION
**Pull request by cacheung**
- Add namespace support for Android Gradle 8.x requirement in plugins
- Update Flutter SDK support version to 3.x
- Update compileSdk to 34
- Update Sample App to use Grade 8.x
- Update GitHub issue_template
- Update plugins release version to 4.0.1 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    description: Please provide a summary of the bug you are reporting.
+  validations:
+    required: true
+
+- type: textarea
+  attributes:
+    label: Environment
+    description: |
+      Please provide the Platform, OS version, SDK version(s) used, IDE version, and any other specific settings that could help us narrow down the problem.
+      Example:
```
Comment (yangyansong-adbe) [inline .github/ISSUE_TEMPLATE/bug_report.yml:30]: Can we also suggest customers add the output of the `flutter doctor -v` command to the bug report?

**Comment by yangyansong-adbe** [comment]
@cacheung 
https://github.com/adobe/aepsdk_flutter/blob/main/plugins/flutter_aepassurance/pubspec.lock.
I suggest that we delete the `pubspec.lock` files present in each plugin directory and ignore them in `.gitignore`.  The dependency lock file is usually controlled by the app developer, we should not include it in the plugin projects. 
cc @cdhoffmann @praveek

**Review by yangyansong-adbe — APPROVED** [review]

File: .github/workflows/dart.yml [hunk 2]
```diff
@@ -44,14 +44,14 @@ jobs:
 
       - uses: maxim-lobanov/setup-xcode@v1
         with:
-          xcode-version: "15.0.1"
+          xcode-version: "15.2"
```
Comment (cdhoffmann) [inline .github/workflows/dart.yml:47]: why are we changing this? We should build on the earliest version we support.

File: plugins/flutter_aepassurance/android/build.gradle
Comment (cdhoffmann) [inline plugins/flutter_aepassurance/android/build.gradle:1]: Need to update the versions for all the plugins to 3

**Review by cdhoffmann — CHANGES_REQUESTED** [review]

**Review by dsoffiantini — APPROVED** [review]

**Review by cdhoffmann — APPROVED** [review]

**Pull request merged**